src/main/java/caca/Storage.java

```diff
@@ -0,0 +1,189 @@
+package caca;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+
+import caca.exceptions.InvalidDateException;
+import caca.tasks.Deadline;
+import caca.tasks.Event;
+import caca.tasks.Task;
+import caca.tasks.Todo;
+
+/**
+ * This class represents the storage system to save and read tasks from the hard disk.
+ *
+ * @author Carrie Zheng Jiarui
+ * @version CS2103T AY22/23 Semester 1, iP
+ */
+public class Storage {
+
+    /**
+     * A string representing the directory path.
+     */
+    private final String dirPath;
+
+    /**
+     * A string representing the file path.
+     */
+    private final String filePath;
+
+    /**
+     * Constructor for creating a Storage.
+     *
+     * @param dirPath Directory path of file being read.
+     * @param filePath Path of the file being read.
+     */
+    public Storage(String dirPath, String filePath) {
+        this.dirPath = dirPath;
+        this.filePath = filePath;
+    }
+
+    /**
+     * Either displays all tasks stored in file (when applicable) as soon as the program starts,
+     * or prompts user if file does not exist.
+     *
+     * @throws FileNotFoundException If file is not found.
+     */
+    public void loadFile() throws FileNotFoundException {
+        String line = "____________________________________________________________\n";
+
+        File file = new File(filePath);
+
+        if (file.exists()) {
+
+            String openFileMsg = "Opening the file that stores all your tasks...\n\n";
+            String showTaskmsg = "These are the tasks in your list:";
```

Comment:
Could it be possible to let ur Ui class handle all your display messages?

Reply:
I'll try to see how I can improve on that. Thank you :)